docs/src/references/c8y-configuration-management.md

```diff
@@ -0,0 +1,153 @@
+  The aim is to avoid breaking the system with half downloaded files.
+* When a download file is copied to its target, the unix user, group and mod are preserved.  
+* Once a snapshot has been downloaded from Cumulocity to the device,
+  __the plugin publishes a notification message on the local thin-edge MQTT bus__.
+  The device software has to subscribe to these messages if any action is required,
+  say to check the content of file, to preprocess it or to restart a daemon. 
+* In other words, the responsibilities of the plugin are:
+  * to define the list of files under configuration management
+  * to notify the cloud when this list is updated,
+  * to upload these files to the cloud on demand,  
+  * to download the files pushed from the cloud,
+  * to make sure that the target files are updated atomically after successful download,
+  * to notify the device software when the configuration is updated.
+* By contrast, the plugin is not responsible for:
+  * checking the uploaded files are well-formed,
+  * restarting the configured processes.
+* A user-specific component, installed on the device,
+  can implement more sophisticated configuration use-cases by:
+  * listening for configuration updates on the local thin-edge MQTT bus,
+  * restarting the appropriate processes when appropriate,  
+  * declaring intermediate files as the managed files,
+    to have the opportunity to check or update their content
+    before moving them to the actual targets.
+
+## Installation
+
+Assuming the configuration plugin `c8y_configuration_plugin`
+has been installed in `/usr/bin/c8y_configuration_plugin`,
+two files must be added under `/etc/tedge/operations/c8y/`
+to declare that this plugin supports two Cumulocity operations:
+uploading and downloading configuration files
+(which respective SmartRest2 codes are `526` and `524`).
+
+These two files can be created using the `c8y_configuration_plugin --init` option: `
+
+```shell
+$ sudo c8y_configuration_plugin --init
+
+$ ls -l /etc/tedge/operations/c8y/c8y_UploadConfigFile
+-rw-r--r-- 1 tedge tedge 95 Mar 22 14:24 /etc/tedge/operations/c8y/c8y_UploadConfigFile
+  
+$ ls -l /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+-rw-r--r-- 1 tedge tedge 97 Mar 22 14:24 /etc/tedge/operations/c8y/c8y_DownloadConfigFile
+```
+
+The `c8y_configuration_plugin` has to be run as a daemon on the device, the latter being connected to Cumulocity.
+
+## Configuration
+
+The `c8y_configuration_plugin` configuration is stored by default under `/etc/tedge/c8y/c8y-configuration-plugin.toml`
+
+This [TOML](https://toml.io/en/) file defines the list of files to be managed from the cloud tenant.
+
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    '/etc/tedge/tedge.toml',
+    '/etc/tedge/mosquitto-conf/c8y-bridge.conf',
+    '/etc/tedge/mosquitto-conf/tedge-mosquitto.conf',
+    '/etc/mosquitto/mosquitto.conf'
+  ]
```

Comment:
Changed by the commit https://github.com/thin-edge/thin-edge.io/pull/1101/commits/8ba09928c38b13bff946cc0bdf4c3d7ababcd11b.

Now the expected format is:
```toml
[[files]]
    path =  "/etc/tedge/tedge.toml"
[[files]]
    path =  "/etc/tedge/mosquitto-conf/c8y-bridge.conf"
[[files]]
    path =  "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf"
[[files]]
    path =  "/etc/mosquitto/mosquitto.conf"
```

Reply:
I have a slight preference for the below format over the one mentioned above (with repeating `[[files]]` blocks):

```toml
files = [
    { path = '/var/log/mosquitto.log' },
    { path = '/var/log/tedge/agent/software-management/software-*' },
]
```

even though both are **equivalent** as per TOML guidelines.